java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java

```diff
@@ -166,13 +168,23 @@ public void setInitialCapacity(int numRecords) {
    *                This helps in tightly controlling the memory we provision
    *                for inner data vector.
    */
+  @Override
   public void setInitialCapacity(int numRecords, double density) {
+    if ((numRecords * density) >= Integer.MAX_VALUE) {
+      throw new OversizedAllocationException("Requested amount of memory is more than max allowed");
+    }
     offsetAllocationSizeInBytes = (numRecords + 1) * OFFSET_WIDTH;
-    final int innerValueCapacity = (int)(numRecords * density);
-    if (innerValueCapacity < 1) {
-      throw new IllegalArgumentException("With the provided density and value count, potential value capacity for the data vector is 0");
+    int innerValueCapacity = (int)(numRecords * density);
+
+    if(innerValueCapacity == 0) {
+      innerValueCapacity = 1;
+    }
```

Comment:
same as above using `Math.max` to make more compact

Reply:
Done.